services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -852,10 +852,11 @@ public void stopInner() throws Exception {
      * correctly.
      */
 
+    LOGGER.info("Waiting to make sure all in-flight requests are drained");
     // Graceful shutdown: Wait till all the requests are drained
     try {
       RetryUtils.executeWithMaxAttempt(() -> {
-        if (RouterHttpRequestStats.hasInFlightRequests()) {
+        if (hasInFlightRequest()) {
```

Comment:
The default retry delay is 10s, which means the total time taken to stop after draining all the ongoing requests can be 10s + 2 * time windows, which will be 20s, right?
can we reduce the retry delay, but increase the retry attempts? For example reduce the retry delay to 1s, but increase the retry attempts to 60?

Also how much time did you observe in your test case?